src/Native/Bootstrap/main.cpp

```diff
@@ -28,17 +56,31 @@ __declspec(allocate(".modules$Z")) void * __modules_z[] = { nullptr };
 extern "C" void __managedcode_a();
 extern "C" void __managedcode_z();
 
-#else
-// TODO: Once the dotnet compiler is modified to configure the linker to merge sections
-// on OSX / Linux, store these externs in the .modules section and remove work-around
-// in ModuleHeaderNode.cs.
-//extern "C" __attribute((section ("__DATA,.modules$A"))) void * __modules_a[];
-//extern "C" __attribute((section ("__DATA,.modules$Z"))) void * __modules_z[];
-//__attribute((section ("__DATA,.modules$A"))) void * __modules_a[] = { nullptr };
-//__attribute((section ("__DATA,.modules$Z"))) void * __modules_z[] = { nullptr };
+#else // _MSC_VER
+
+#if defined(__APPLE__)
+
+//
+// On platforms with MachO binaries (OSX), we cannot use the above mechanism to register
+// global data for each module. Instead, we use a special section type whose contents
+// are a list of function pointers to call.  We create a jump stub that passes the global
+// data pointer into RegisterReadyToRunModule where we save it in a list for the startup
+// code to consume.
+//
+std::vector<void*> __registeredModules;
```

Comment:
It would be nice to not introduce dependency on C++ STL - it pulls in a pile of stuff that we do not have/need.